### PR TITLE
refactor: Created a new header-layout file to segregate out header code.

### PR DESCRIPTION
**Pull request by Mohitsen11**
## Summary

This PR introduces modular approach by separating header-code in a new 'header-layout' file  rather than keeping into a single file (home-page).

## Description

This PR isolates the header-related code from the home-page file and moves it into a new header-layout file, along with all imported modules. It then integrates this header-layout as a component within the home-page file.

## Images

_Include any relevant images or diagrams that can help reviewers visualize the changes, if applicable_

## Issue(s) Addressed

_Enter the issue number of the bug(s) that this PR fixes_

- Template should be strictly **Closes <issue_number>**
- Example: Closes #1

## Prerequisites

- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

File: frontend/src/layouts/header-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import ThemeToggle from '@/components/theme-toggle-button';
+import AddIcon from '@/assets/svg/add-icon-white.svg';
+import { useNavigate } from 'react-router-dom';
+function header() {
+    const navigate = useNavigate();
+
+    return (
+    <div
+        style={{ backgroundImage: `url(${bg})` }}
+        className="relative -mt-2 h-[460px] bg-cover bg-fixed bg-center"
```
Comment (krishnaacharyaa) [inline frontend/src/layouts/header-layout.tsx:11]: Can you kindly refactor this to use bg-[url] kinda format, the tailwind supported one

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Please create new branch and cherry-pick , because it is on top of the previous merged branch.
And can you just keep till the Add post in the header and the rest "Dive .. " part create a HeroComponent

File: frontend/src/layouts/header-layout.tsx (hunk 1 above)
Comment (Mohitsen11) [reply frontend/src/layouts/header-layout.tsx:11]: Done everything.

Comment (Mohitsen11) [reply frontend/src/layouts/header-layout.tsx:11]: Created a PR with new branch.

**Comment by krishnaacharyaa** [comment]
Declining as #124 duplicate

**Pull request closed without merging**